ArticleTemplates/assets/js/modules/util.js

```diff
@@ -41,6 +41,34 @@ define(function () {
 
         isOnline: function () {
             return !document.body.classList.contains('offline') && navigator.onLine;
+        },
```

Comment:
At risk of building your own qwery, I wonder whether a `each` util might be useful to clear out some of the for loops and make the code easier to reason about?